### PR TITLE
feat: parsing composition

### PR DESCRIPTION
**Pull request by MightyZanark**
This PR adds parsing for composition, which, since KenKomKom has added most of the ground work, nothing much have been added other than conditionals for composition.

Completes MOT-157

### CODE REVIEW

File: app/parse_class_pattern/parse_relationship_strategy.py [hunk 1]
```diff
@@ -138,6 +160,17 @@ def create_relationship(
...
+        # specific for django implementation of composition for now
+        # since "end" is the class that "owns" the "start" class but
+        # in django the "owned by" class needs to be the one to set FK
+        if ro.get_type() == RelationshipType.COMPOSITION:
+            ro.set_source_class(class_to_id)
+            ro.set_target_class(class_from_id)
+            ro.set_source_class_own_amount(edge["endLabel"])
+            ro.set_target_class_own_amount(edge["startLabel"])
+            class_to_id.add_relationship(ro)
+            return
```
[the remark this answers is not in the document]
Comment (MightyZanark) [reply app/parse_class_pattern/parse_relationship_strategy.py:155]: Refactor another way

File: app/parse_class_pattern/parse_relationship_strategy.py [hunk 2]
```diff
@@ -102,6 +113,17 @@ def create_relationship(
...
+            # specific for django implementation of composition for now
+            # since "end" is the class that "owns" the "start" class but
+            # in django the "owned by" class needs to be the one to set FK
+            if ro.get_type() == RelationshipType.COMPOSITION:
+                ro.set_source_class(class_to_id)
+                ro.set_target_class(class_from_id)
+                ro.set_source_class_own_amount(edge["endLabel"])
+                ro.set_target_class_own_amount(edge["startLabel"])
+                class_to_id.add_relationship(ro)
+                return
```
[the remark this answers is not in the document]
Comment (MightyZanark) [reply app/parse_class_pattern/parse_relationship_strategy.py]: Refactored another way

File: app/parse_class_pattern/parse_relationship_strategy.py [hunk 3]
```diff
@@ -43,6 +43,17 @@ def create_relationship(
...
+        # specific for django implementation of composition for now
+        # since "end" is the class that "owns" the "start" class but
+        # in django the "owned by" class needs to be the one to set FK
+        if ro.get_type() == RelationshipType.COMPOSITION:
+            ro.set_source_class(class_to_id)
+            ro.set_target_class(class_from_id)
+            ro.set_source_class_own_amount(edge["endLabel"])
+            ro.set_target_class_own_amount(edge["startLabel"])
+            class_to_id.add_relationship(ro)
+            return
```
[the remark this answers is not in the document]
Comment (MightyZanark) [reply app/parse_class_pattern/parse_relationship_strategy.py]: Refactored another way

File: app/parse_class_pattern/parse_relationship_strategy.py [hunk 4]
```diff
@@ -138,6 +143,17 @@ def create_relationship(
...
```
Comment (KingOnyxx) [inline app/parse_class_pattern/parse_relationship_strategy.py:155]: i think you can refactor this into using strategy pattern

File: tests/test_parse_class_pattern/test_strategy.py [hunk 5]
```diff
@@ -126,6 +126,66 @@ def test_many_to_one_aggregation_invalid(self):
...
+
+        strategy.create_relationship(
+            edge, cls_src, cls_target, RelationshipType.COMPOSITION
+        )
+
+        relationships = cls_target._ClassObject__relationships
+        self.assertEqual(len(relationships), 1)
+        self.assertEqual(relationships[0].get_type(), RelationshipType.COMPOSITION)
+        self.assertEqual(type(relationships[0]), ManyToManyRelationshipObject)
+
```
Comment (KingOnyxx) [inline tests/test_parse_class_pattern/test_strategy.py:188]: Should there also have more negative test case or edge case to check how the code behaves?

**Review by KingOnyxx — APPROVED** [review]
Thank you for implementing these changes! After reviewing, there are things that could be improved but it can be for future refactoring task. Approved from me!

File: tests/test_parse_class_pattern/test_strategy.py (hunk 5 above)
Comment (MightyZanark) [reply tests/test_parse_class_pattern/test_strategy.py:188]: When the parsing step reaches this point, if the input is still wrong I think it should be the parser's responsibility for missing or unable to identify it. The other cases for the `create_relationship` are already handled on other methods.

File: app/parse_class_pattern/parse_relationship_strategy.py (hunk 4 above)
Comment (MightyZanark) [reply app/parse_class_pattern/parse_relationship_strategy.py:155]: I have refactored this another way on the other PR.
